src/signer/canister/src/sign/bitcoin/tx_utils.rs

```diff
@@ -140,12 +140,17 @@ pub async fn build_p2wpkh_transaction(
     }
 }
 
+fn is_same_outpoint(txin_outpoint: &BitcoinOutPoint, utxo_outpout: &IcCdkOutPoint) -> bool {
+    txin_outpoint.vout == utxo_outpout.vout
+        && txin_outpoint.txid.as_byte_array().to_vec() == utxo_outpout.txid
```

Comment:
Comparing as slices avoids creating a vector, so is faster:
```suggestion
        && txin_outpoint.txid.as_byte_array()[..] == utxo_outpout.txid[..]
```